determined-ai/determined
# fix: reduce the number of api calls from Workspace Create Modal

## Ticket

DET-10443


## Description

Workspace Create Modal was calling 3 API's multiple times - which was slowing down the webUi severely. Reduced the number of calls and made the code more efficient.


## Test Plan

Check for both Agent and Kubernetes RM: 
- Open the workspace list page as a non admin user and check that there are no failing API calls for getKubernetesResourceManagers, ListNamespaceBindings and GetK8sResourceQuotas. 
- Open the workspace create/Edit modal from multiple places and verify it renders correctly and only getKubernetesResourceManagers API call fails once without a toast. 
- Login as admin and make sure none of these API calls fail

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
